Final Project - Print shop - Jenny A

jempa182 opened the pull request:
Frontend: https://jenny-a.netlify.app/
Backend: https://jenny-a-artwork.onrender.com/prints

jempa182 committed:
setup project structure and installs

jempa182 committed:
added endpoints

jempa182 committed:
added basic styling to product page and product detail page

jempa182 committed:
added cart function, profile page, login and logout

jempa182 committed:
update

jempa182 committed:
test

jempa182 committed:
added pagination, hover effects and header

jempa182 committed:
like button function + custom font

jempa182 committed:
filter categories

jempa182 committed:
stripe and categories

jempa182 committed:
netlify toml file and vite config file update

jempa182 committed:
mongoDB env file

jempa182 committed:
updated updated backend links

jempa182 committed:
added env file

JennieDalgren commented on backend/config/database.js:
```diff
@@ -0,0 +1,22 @@
+// Database configuration
+const config = {
+  // Added comment to remind about moving to .env
+  url: process.env.MONGODB_URL || "mongodb+srv://jempa182:NjzFCota9q0e9Oy8@cluster0.1aoha.mongodb.net/project-mongodb?retryWrites=true&w=majority&appName=Cluster0"
```
Do not expose you cluster like this, an env variable should be enough.

JennieDalgren commented on backend/config/stripe.js:
```diff
@@ -0,0 +1,19 @@
+// Stripe configuration
+const config = {
+  // Added comment to remind about moving to .env
+  secretKey: process.env.STRIPE_SECRET_KEY || 'sk_test_51Qjh2QAo9otVE0iMVDhH43JhXiWLjz4fHD1EbAkkccnw0OObDymX92FxnojyB79LlcWEgVJRYGv2Qz2Kr87WFNMt00WNrDetDE',
+  webhookSecret: process.env.STRIPE_WEBHOOK_SECRET || 'whsec_edf4e79bb58330085a6aefab4154850f9ead6ef724a21e78f3cae338488b58c8',
+  currency: 'sek',
+  shipping: {
+    cost: 49,  // Default shipping cost
+  }
```
even if this is just a test account, do not expose your stripe keys here

JennieDalgren commented on backend/data/seedPrints.js:
```diff
@@ -0,0 +1,199 @@
+// backend/data/seedPrints.js
```
I understand that you might want to keep your prints in a json file like this as a backup. But once you have seeded the database, you can remove this file from the project to make it more clean and tidy

JennieDalgren commented on backend/models/Print.js:
```diff
@@ -0,0 +1,36 @@
+  category: {
+    type: String,
+    required: true,
+    enum: ['Art Prints', 'T-shirts', 'Accessories', 'Stickers']
```
⭐

JennieDalgren commented on backend/routes/orders.js:
```diff
@@ -0,0 +1,106 @@
+  try {
+    const { items, shippingAddress } = req.body;
+    
+    // Calculate total with shipping
+    const totalAmount = items.reduce((sum, item) => sum + (item.price * item.quantity), 0);
+    const shippingCost = 49;
+    const finalAmount = totalAmount + shippingCost;
+
```
here I would check if the quanitity is 0, to remove it from the cart or at least de-activate the Checkout button. Either take care of this here and/or in the frontend. 
Screenshot for reference:
<img width="692" alt="image" src="https://github.com/user-attachments/assets/de9490d5-5658-4994-a9cd-8a69fcb0f439" />

JennieDalgren reviewed (dismissed):
Great job with this project! It looks so professional and I love the design. 
It would be even more nice to read an actual about text and see your prints and t-shirts! However that is not necessary for passing this project.

But there are some things that needs fixing,
The favorites section doesnt really work as expected. When I "like" items they get marked with a red heart in the shopping feed as well as listed in My Favorites. But in the profile view it looks like i dont have any faves.

<img width="774" alt="image" src="https://github.com/user-attachments/assets/c13d0937-0fa0-4855-b21c-cc226096c3b8" />
<img width="774" alt="image" src="https://github.com/user-attachments/assets/8e45114b-3039-46eb-a6c5-149adb513d36" />

Also, if I select a print to the cart, and then go to the cart and change the amount to 0 I would assume that I would not be able to checkout, only paying for the shipping but no products. 😄 

Also, the Stripe checkout errors out - 
<img width="1410" alt="image" src="https://github.com/user-attachments/assets/cdffe69d-15e3-4c50-899e-e2f0f3029327" />
If this wasn't implemented fully, maybe let the user know somehow.

Please take a look at the 
- favorites not showing in profile
- Actual checkout issue
- Empty cart should be able to get bought.

Keep it up! 💪

jempa182 committed:
fixed favorites count on profile page

jempa182 committed:
fixed updating cart to 0 issue

HIPPIEKICK commented:
Hi Jenny!
Just wanted to let you know that I checked your project changes, looks really good! The only thing left now would be the Stripe error in the console and the inline comments Jennie made in your code (e.g. about env variables).

Almost there!

jempa182 committed:
changes to guest checkout payment

jempa182 committed:
updates orders.js

jempa182 committed:
fix error in database.js

jempa182 committed:
env file updates

jempa182 committed:
error handling payment issue

jempa182 committed:
checkout page issues test

jempa182 committed:
test checkout page

jempa182 committed:
test checkout error handling

jempa182 committed:
orderconfirmation test

jempa182 committed:
added netlify.toml

jempa182 committed:
order confirmation test button

jempa182 committed:
removed error handling and fallbackKey for testing

jempa182 committed:
tidy up comments

jempa182 commented:
Hi @HIPPIEKICK! Thanks a lot 😊 I think I have now fixed the checkout error issue. Also removed the stripe keys so they should hopefully not be visible anywhere now. Have a look and let me know if it looks correct! 😄
